index.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const assert = require('assert');
+
+class DejaVu {
+
+  /**
+   * Constructs an new instance of DejaVu to identify "original" events.
+   * @param {Object} options The constructor options.
+   * @param {Object} options.redisConnection A connection to a Redis deployment.
+   */
+  constructor(options) {
+    assert(options.redisConnection, 'Must provide a Redis connection');
+    this._redisConnection = options.redisConnection;
+    this._handlers = {};
+  }
+
+  /**
+   * isNew returns if an event is new or not.
+   * @param {String} prefix The prefix to namespace events under.
+   * @param {Number} window The window to filtet event timestamps by (in ms).
+   * @param {Function} timestampFn A function for extracting an event's timestamp.
+   * @param {Function} idFn A function for extracting an event's ID.
+   * @param {Object} eve The event of interest.
+   * @param {Function} done Node style callback.
+   */
+  isNew(prefix, window, timestampFn, idFn, eve, done) {
+    const now = Date.now();
+    const occurredAt = timestampFn(eve);
+    if (!occurredAt) return done(null, false);
```

Comment:
I'll return an error instead, as I don't think we should allow events in that have a different schema.